tests/phpunit/tests/AMP/Output_Buffer.php

```diff
@@ -0,0 +1,237 @@
+<?php
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Tests\AMP;
+
+use Google\Web_Stories\Experiments;
+use Google\Web_Stories\Model\Story;
+use Google\Web_Stories\Renderer\Story\HTML;
+use Google\Web_Stories\Settings;
+use Google\Web_Stories\Story_Post_Type;
+use Google\Web_Stories\Tests\Test_Case;
+
+/**
+ * @coversDefaultClass \Google\Web_Stories\AMP\Output_Buffer
+ */
+class Output_Buffer extends Test_Case {
+	public function setUp() {
+		parent::setUp();
+
+		// When running the tests, we don't have unfiltered_html capabilities.
+		// This change avoids HTML in post_content being stripped in our test posts because of KSES.
+		remove_filter( 'content_save_pre', 'wp_filter_post_kses' );
+		remove_filter( 'content_filtered_save_pre', 'wp_filter_post_kses' );
+	}
+
+	public function tearDown() {
+		add_filter( 'content_save_pre', 'wp_filter_post_kses' );
+		add_filter( 'content_filtered_save_pre', 'wp_filter_post_kses' );
+
+		parent::tearDown();
+	}
+
+	protected function prepare_response( $post ): string {
+		$instance = new \Google\Web_Stories\AMP\Output_Buffer( new Experiments() );
```

Comment:
Should Experiments be mocked here?m

Reply:
Maybe, doesn't really make a difference. Also, this code was pretty much just moved 1:1 from the HTML class tests